### PR TITLE
Improve how to serialize fulfilled promise status

### PR DESCRIPTION
**Pull request by CYBAI**
#### 359976cda05e9714df3462327e2b3d0fc8527649
<pre>
Improve how to serialize fulfilled promise status
<a href="https://bugs.webkit.org/show_bug.cgi?id=225328">https://bugs.webkit.org/show_bug.cgi?id=225328</a>

Reviewed by Patrick Angle.

While the `[[PromiseStatus]]` enum matches the spec to have `fulfilled`
status, WebKit is translating it to `resolved` which doesn&apos;t match the
spec[1]. Thus, this commit will make `fulfilled` status to be serialized
as `fulfilled` instead of `resolved`.

[1]: <a href="https://tc39.es/ecma262/#sec-properties-of-promise-instances">https://tc39.es/ecma262/#sec-properties-of-promise-instances</a>

* LayoutTests/inspector/model/remote-object/promise-expected.txt:
* LayoutTests/inspector/runtime/getDisplayableProperties-expected.txt:
* LayoutTests/inspector/runtime/getProperties-expected.txt:
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
(Inspector::JSInjectedScriptHost::getInternalProperties):

Canonical link: <a href="https://commits.webkit.org/266058@main">https://commits.webkit.org/266058@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/7e3d3cf1cc46a825c12ce2aa39314fd8abd2f776

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11843 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12042 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12414 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13488 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11280 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14431 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12023 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14138 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12007 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12834 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10035 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13910 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10126 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10754 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17882 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/10051 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11204 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10909 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14077 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/11214 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11318 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9355 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11925 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10493 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3207 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3119 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14776 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/12261 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11174 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2973 "Passed tests") |

### CODE REVIEW

**Comment by rcaliman-apple** [comment]
This seems like a sensible change to bring Safari Web Inspector in line with Chrome DevTools and Firefox DevTools in the string they present in the Console:


Screenshot 2023-07-05 at 20 08 53
Screenshot 2023-07-05 at 20 09 05
Screenshot 2023-07-05 at 20 09 57



@JosephPecoraro @dcrousso  Do you foresee any side-effects from this change in `Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp`?

**Comment by dcrousso** [comment]
i personally prefer `"resolved"` as that better matches my experience using JS (e.g. `Promise.resolve(...)`, `new Promise((resolve, reject) => { ... })`, etc.)

also, i dont think we need to be following/exposing the terminology used by the spec as that's easily changeable since it's not exposed to developers (e.g. im pretty sure it used to be `"resolved"` and was changed at some point to be `"fulfilled"`)

i dont feel super strongly though so if others do then i am happy to yield

**Comment by CYBAI** [comment]
Sorry, I should ask in the bug or on Slack first next time 🙏
If the team would like to keep `"resolved"`, I will close this PR 🙇

**Review by patrickangle — APPROVED** [review]
While aligning such a small detail with Chrome/Firefox for the sake of aligning isn't a real concern to me, and while aligning 100% with the evolving spec language also isn't a big concern (since if it changed once, it can change again, and perhaps confuse users as it changes on our end), it seems to me that both factors combined are a good enough reason to seek alignment with **both** the spec and other vendors to make the experience more familiar as developers move across platforms and engines.

**Comment by Ahmad-S792** [comment]
@CYBAI - Do you want any committer to add 'merge-queue'?

**Comment by CYBAI** [comment]
Sorry for late reply 🙏  Yes! and thanks @rcaliman-apple for helping me add the `merge-queue` label 🙇

**Comment by rcaliman-apple** [comment]
> Sorry for late reply 🙏 Yes! and thanks @rcaliman-apple for helping me add the `merge-queue` label 🙇

No worries! I should have done this earlier as soon as the PR got approved.

**Pull request merged**
